Sources/DataSource.swift

```diff
@@ -28,6 +28,7 @@ import ReactiveKit
 public protocol DataSourceProtocol {
   var numberOfSections: Int { get }
   func numberOfItems(inSection section: Int) -> Int
+  func item<U>(at index: Int) -> U
```

Comment:
I hate adding this requirement to `DataSourceProtocol` because it's not always need. Also, the function should not be generic, rather we should introduce associated types for Item and Index (it is not alway Int). I suggest another protocol that inherits this one:

```swift
public protocol QueryableDataSourceProtocol: DataSourceProtocol {
  associatedtype Item
  associatedtype Index
  func item(at index: Index) -> Item
}
```

If you have a better name that would be cool :)